index.d.ts

```diff
@@ -73,7 +73,7 @@ export interface BadgeProps {
      *
      * @default View (if onPress then TouchableOpacity)
      */
-    Component?: React.ComponentClass;
+    Component?: React.ComponentType<T>;
```

Comment:
If <T> is used a type param, you need to add it as a type param of BadgeProps as well

Reply:
I can't commit those change from the UI, can you?

Reply:
Not sure what you meant, what I was trying to say was that adding <T> like this is not enough. The <T> should come from BadgeProps<T> as well.